File: surface/widgets/swiftui/components/widgets/composites/CacheDashboard.swift
// ============================================================
// Clef Surface SwiftUI Widget — CacheDashboard
//
// Dashboard displaying cache statistics, hit/miss rates, and
// cache management actions (clear, refresh).
// ============================================================

import SwiftUI

struct CacheEntry: Identifiable {
    let id: String
    let key: String
    let size: String
    var hitCount: Int = 0
    var age: String = ""
}

struct CacheDashboardView: View {
    var entries: [CacheEntry]
    var totalSize: String = "0 KB"
    var hitRate: Double = 0
    var onClear: (() -> Void)? = nil
    var onRefresh: (() -> Void)? = nil

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack {
                Text("Cache Dashboard").font(.headline).fontWeight(.bold)
                Spacer()
                if let onRefresh = onRefresh {
                    SwiftUI.Button(action: onRefresh) { Image(systemName: "arrow.clockwise") }.buttonStyle(.plain)
                }
                if let onClear = onClear {
                    SwiftUI.Button("Clear", role: .destructive, action: onClear)
                }
            }
            HStack(spacing: 16) {
                VStack { Text("Total Size").font(.caption).foregroundColor(.secondary); Text(totalSize).font(.headline) }
                VStack { Text("Hit Rate").font(.caption).foregroundColor(.secondary); Text("\(Int(hitRate * 100))%").font(.headline) }
                VStack { Text("Entries").font(.caption).foregroundColor(.secondary); Text("\(entries.count)").font(.headline) }
            }
            Divider()
            ForEach(entries) { entry in
                HStack {
                    VStack(alignment: .leading) {
                        Text(entry.key).font(.subheadline).fontWeight(.medium)
                        Text("\(entry.size) | \(entry.hitCount) hits | \(entry.age)").font(.caption).foregroundColor(.secondary)
                    }
                    Spacer()
                }
                .padding(.vertical, 4)
            }
            if entries.isEmpty { Text("Cache is empty").font(.body).foregroundColor(.secondary) }
        }.padding(12)
    }
}

File: surface/widgets/nativescript/components/widgets/composites/CacheDashboard.ts
// ============================================================
// Clef Surface NativeScript Widget — CacheDashboard
//
// Cache metrics dashboard with hit/miss rates and key browser.
// ============================================================

import { StackLayout, Label, Button, ScrollView } from '@nativescript/core';
import type { View } from '@nativescript/core';

export interface CacheDashboardProps {
  children?: View[];
  [key: string]: unknown;
}

export function createCacheDashboard(props: CacheDashboardProps): StackLayout {
  const { children = [], ...rest } = props;
  const container = new StackLayout();
  container.className = 'clef-widget-cache-dashboard';
  container.accessibilityLabel = 'Cache Dashboard';

  const header = new Label();
  header.text = 'Cache Dashboard';
  header.fontWeight = 'bold';
  header.fontSize = 16;
  container.addChild(header);

  for (const child of children) container.addChild(child);
  return container;
}

export default createCacheDashboard;
